climt/_components/rrtmg/lw/component.py

```diff
@@ -248,16 +252,37 @@ def __init__(
         else:
             self._calc_dflxdt = 0
 
+        self._mcica = mcica
+        if mcica:
+            self._random_number_generator = random_number_generator
+            self._permute_seed = permute_seed
+            if type(cloud_overlap_method) is str:
+                if cloud_overlap_method.lower() == 'clear_only':
```

Comment:
You have added a lot of `if` statements, which will reduce coverage by quite a bit. You can add one test (just a function like `test_rrtmg_logging`) which will cover these additional lines. In this test/s you should initialise RRTMG with the option which will trigger these logging statements and capture them using pytest:
https://docs.pytest.org/en/latest/logging.html
something like 
```python
def test_baz(caplog):
    func_under_test()
    for record in caplog.records:
        assert record.levelname != 'CRITICAL'
    assert 'wally' not in caplog.text
```
from the above docs.